.github/scripts/rust/lint.rs

```diff
@@ -0,0 +1,255 @@
+        panic!(".cargo/config.toml needs to exist")
+    }
+
+    let lints = collect_lints(cwd);
+
+    // read the `config.toml`
+    let contents = fs::read_to_string(&path).expect(&format!("should be able to read {}", path.display()));
+
+    if !contents.contains("## START CLIPPY LINTS ##") || !contents.contains("## END CLIPPY LINTS ##") {
+        panic!("malformed .cargo/config.toml, please add the required markers")
+    }
+
+    // strip out content between the markers
+    let prefix = contents.lines().take_while(|value| value.trim() != "## START CLIPPY LINTS ##");
+    let suffix = contents.lines().skip_while(|value| value.trim() != "## END CLIPPY LINTS ##");
+
+    let indent = "    ";
+    let body = lints.allow.into_iter().map(|lint| format!(r#"{indent}"-A{lint}","#))
+        .chain(lints.warn.into_iter().map(|lint| format!(r#"{indent}"-W{lint}","#)))
+        .chain(lints.deny.into_iter().map(|lint| format!(r#"{indent}"-D{lint}","#)))
+        .chain(lints.forbid.into_iter().map(|lint| format!(r#"{indent}"-F{lint}","#)));
+
+    let contents: Vec<_> = prefix.map(|line| line.to_owned())
+        .chain(once(format!("{indent}## START CLIPPY LINTS ##")))
+        .chain(body)
+        .chain(suffix.map(|line| line.to_owned()))
+        .collect();
+
+    fs::write(&path, contents.join("\n")).expect("able to write `.cargo/config.toml`");
+}
+
+fn print_diff(level: &str, left: &HashSet<String>, right: &HashSet<String>) {
+    let create = left - right;
+    let remove = right - left;
+
+    eprintln!("Lint Level {level} Errors:");
+    eprintln!("  Extra lints: {:?}", remove);
+    eprintln!("  Missing lints: {:?}", create);
+    eprintln!("  Run `cargo make update-clippy update` to automatically update the config.")
+}
+
+fn check(cwd: &Path) {
+    let lints = collect_lints(cwd);
+    let cargo = collect_cargo(cwd);
+
+    let mut fail = false;
+
+    if !(&lints.allow ^ &cargo.allow).is_empty() {
+        fail = true;
+        print_diff("Allow", &lints.allow, &cargo.allow);
+    }
+
+    if !(&lints.warn ^ &cargo.warn).is_empty() {
+        fail = true;
+        print_diff("Warn", &lints.warn, &cargo.warn);
+    }
+
+    if !(&lints.deny ^ &cargo.deny).is_empty() {
+        fail = true;
+        print_diff("Deny", &lints.deny, &cargo.deny);
+    }
+
+    if !(&lints.forbid ^ &cargo.forbid).is_empty() {
+        fail = true;
+        print_diff("Forbid", &lints.forbid, &cargo.forbid);
+    }
+
+    if fail {
+        std::process::exit(1);
+    } else {
+        println!("No changes necessary, we're good to go!");
+    }
+}
+
+fn main() {
+    let cwd = env::var("CARGO_MAKE_WORKSPACE_WORKING_DIRECTORY").expect("environment variable should exist");
+    let cwd = PathBuf::from(cwd);
+
+    let args = env::var("CARGO_MAKE_TASK_ARGS").expect("environment variable should exist");
+    let mode = args.split(" ").next().expect("at least one position argument specifying mode required");
+
+    match mode {
+        "apply" => apply(&cwd),
```

Comment:
Can we run these as dependency on the clippy task?

Reply:
added in [8febc14](https://github.com/hashintel/hash/pull/2075/commits/8febc14322b2212a2e0276c2ebb9140290eb63a4)